### ISSUE TITLE
photos_add page not finding ImageMagick, returns 'GD library is missing'.

### ISSUE THREAD

**Issue by ronankeane** [Piwigo/Piwigo#957]
Newly installed, Piwigo 2.9.4
ImageMagick 7.0.8-23 was properly installed _after_ the install of Piwigo.
(GD Library was never installed on this Ubuntu system).
All other image functionality works fine in this Piwigo installation.

**Comment by flop25** [comment]
@plegall why is there that error reporting https://github.com/Piwigo/Piwigo/blob/master/admin/include/photos_add_direct_prepare.inc.php#L176 while there is a check there https://github.com/Piwigo/Piwigo/blob/master/admin/include/photos_add_direct_prepare.inc.php#L36
Is GD library really mandatory? I only remember that GD is used if the pictures are resized after upload

**Comment by plegall** [comment]
I remember GD is mandatory to read EXIF metadata. But I may be wrong!

**Comment by flop25** [comment]
it is to keep metadata when resizing
but https://github.com/Piwigo/Piwigo/blob/b9336d7117770e304197a3afad1068eea5491a29/include/functions_metadata.inc.php#L141 http://php.net/manual/fr/function.exif-read-data.php (btw php 7.2 improved the readibility of exfis data nice!)

**Comment by ronankeane** [comment]
I notice that there was once a vulnerability in Imagemagick that would be exploited during the photo_add step - https://www.theregister.co.uk/2016/05/03/imagemagick/

**Comment by plegall** [comment]
> it is to keep metadata when resizing
I don't think so. I've made some search on the forum and I said a while back that GD was removing metadata when resizing.

Maybe GD is not really mandatory. What's mandatory is to have either GD or imagemagick. I've always considered GD was coming automatically with PHP. Here is maybe my mistake because today you can perfectly install PHP without GD and have ImageMagick which should be fine.

I'll make some tests without GD and only ImageMagick and if everything works fine, I'll change the condition from "is GD installed" to "is GD or ImageMagick installed".

**Comment by plegall** [comment]
As found by @Teatek in #976 it's possible to use Piwigo without GD, but it will fail for GIF files (animated or not). We need to change some code to adapt ImageMagick command lines to handle static and animated gif files. As said by @mistic100 on https://piwigo.org/forum/viewtopic.php?pid=151293#p151293 and mentioned by Teatek, I agree we would need an option to disable animation on small resized GIF (maybe reusing `$conf['derivatives_strip_metadata_threshold']`)

**Comment by ronankeane** [comment]
I got around to installing the GD library to make the photos_add page work. A couple of issues.

1. While the photos_add page now works fine, it displays a message --- Warning: count(): Parameter must be an array or an object that implements Countable in /var/www/photos/piwigo/_data/templates_c/p6jhns^561a9bf60ffeeda8e86a99ad26c0ec6395fd8cf0_0.file.photos_add_direct.tpl.php on line 327
(note my userlist.tpl is updated as per fix https://github.com/Piwigo/Piwigo/issues/887)

2. I specified $conf['graphics_library'] = 'imagick'; in local config file, but (a) having read your documentation I am unclear as to correct use of 'imagick' or 'ext_imagick', (b) I do not see where I can check and confirm which library is actually being used.
